x/nft/msg.go

```diff
@@ -0,0 +1,56 @@
+package nft
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+var _ weave.Msg = (*AddApprovalMsg)(nil)
+var _ weave.Msg = (*RemoveApprovalMsg)(nil)
+
+const (
+	pathAddApproval    = "nft/add-approval"
+	pathRemoveApproval = "nft/remove-approval"
+)
+
+// Path returns the routing path for this message
+func (*AddApprovalMsg) Path() string {
+	return pathAddApproval
+}
+
+// Path returns the routing path for this message
+func (*RemoveApprovalMsg) Path() string {
+	return pathRemoveApproval
+}
+
+func (m *AddApprovalMsg) Validate() error {
+	if err := weave.Address(m.Address).Validate(); err != nil {
+		return err
+	}
+	if m.Action == "" {
+		return errors.ErrInternal("action must not be empty")
+	}
+	//TODO: Figure out if we care, as we have a default in ops now that adds infinite count
+	if m.Options == nil {
+		return errors.ErrInternal("options must not be nil")
+	}
+	//TODO: Figure out whether we need to incorporate same check as in NFT
+	if len(m.Id) == 0 {
```

Comment:
req: as this is the nft id, the pattern/min/max of the base nft id definitively applies here for fail fast.

Reply:
Yes, that was my plan